pkg/eventqueue/eventqueue.go

```diff
@@ -244,6 +248,7 @@ func (q *EventQueue) Run() {
 				// already been processed.
 				ev.printStats(q)
 				close(ev.eventResults)
+				q.inflightEventMu.Unlock()
 			}
 		}
```

Comment:
[0] It's safer if you add a `defer close(q.eventsClosed)` on line 233 and on line 294 you add `<-q.eventsClosed`

Reply:
You're right, this is much cleaner.